browser/ui/webui/brave_rewards_ui.cc

```diff
@@ -91,6 +109,29 @@ void RewardsDOMHandler::OnWalletCreateFailed() {
   }
 }
 
+  void RewardsDOMHandler::OnWalletProperties(
+      payments::PaymentsService* payment_service,
+      ledger::WalletInfo result) {
+    OnWalletProperties(result);
+  }
+
+void RewardsDOMHandler::OnWalletProperties(ledger::WalletInfo result) {
+  if (0 != (web_ui()->GetBindings() & content::BINDINGS_POLICY_WEB_UI)) {
+    auto* web_contents = web_ui()->GetWebContents();
+    if (web_contents) {
+      auto* render_view_host = web_contents->GetRenderViewHost();
+      if (render_view_host) {
+        render_view_host->SetWebUIProperty("rewards.walletInfo.probi", result.probi_);
+        render_view_host->SetWebUIProperty("rewards.walletInfo.balance", std::to_string(result.balance_));
+        for (auto const& rate : result.rates_) {
+          render_view_host->SetWebUIProperty("rewards.walletInfo.rates." + rate.first , std::to_string(rate.second));
+        }
+      }
+    }
+    web_ui()->CallJavascriptFunctionUnsafe("brave_rewards.walletProperties");
```

Comment:
wouldn't this be a generic method to update the state with the new webui props?

Reply:
we can also keep the balance update timers in c++ and just update the props

Reply:
we talked about this with @bbondy and we decided to have all timers in app and only do calls to library when needed

Reply:
well, I'm not even a big fan of the fact that the balance update runs continuously when the payments ui is active and we definitely don't want it running all the time. I think it probably makes  sense to put that inside the webcontents observer though and not JS

Reply:
balance is running only when you are on rewards page (chrome://rewards). We are already quite smart about it, so for example when you claim grant we call balance etc, so maybe we can have this conversation again and increase timer from a minute to something higher